### PR TITLE
Updates per conversation in 1-on-1 meeting

### PR DESCRIPTION
**Pull request by gilzow**
Adds `REPOOWNER` and `TEMPLATEOWNER` env vars to track github namespaces for repos

Re-orders steps in build (+adds step numbers to names):
- pre-start = all configuration steps for python and poetry, default env vars for passing data between steps
- 1 make sure the template repo exists
- 2 `(new)` verify auto-merge for repo is enabled, if not, enable
- 3 Get repo's default branch name
- 4 `(new)` Add branch protection rule to default branch
- 5 `(new)` Patch branch protection rule to remove review requirement
- 6 Check if update branch already exists
- 7 doit cleanup
- 8 doit initialize template build
- 9 doit initialize remote
- 10 doit update
- 11 doit platformify
- 12 doit branch template
- 13 Push template
- 14 check if updates were pushed to update branch. If not, it's assumed there were no updates. Steps 15 & 16 skipped `(new)`
- 15 Create PR
- 16 Merge the PR with auto-merge enabled `(new)` and delete branch on merge enabled `(new)`
- 17a & b. Send notices to Slack

### CODE REVIEW

**Comment by gilzow** [comment]
commit f603633 :
**Adds**
* env var `PYTHON_TEMPLATES` which contains a list of all python-based templates

**Updates**
* `UPDATES_BRANCH` and `UPDATE_COMMIT_MSG` env vars to use single quotes to fix issue with python being able to access those values
* Update steps checks to see if we're updating a python project, and if so, run a second update command

File: .github/workflows/updates.yaml [hunk 1]
```diff
@@ -158,63 +162,153 @@ jobs:
...
+                DEFAULT_BRANCH=$(gh api /repos/${TEMPLATEOWNER}/${{ matrix.template }} --jq '.default_branch')
+                echo "Our default branch for this template is ${DEFAULT_BRANCH}"
+                echo "::set-output name=branch::$DEFAULT_BRANCH"
+
+            # 4. Check if default branch has branch rules in place
+            # @todo do we bother to check, or do we just assume it isnt set and attempt to create it anyway? If we attempt to create it and it already
+            # exists, it should just return the same values anyway
+            - name: Step 4. Add branch protection rules to default branch
+              run: |
+                # @todo it would be nice if there was some way to verify if the protection rule we need is set and in place instead of always
```
Comment (chadwcarlson) [inline .github/workflows/updates.yaml:243]: If this helps https://gist.github.com/chadwcarlson/1938d8619ebe7338e91d720b06685a26 @gilzow

**Comment by gilzow** [comment]
Commit 1b60ecf 
Eliminates steps 4 (create branch protection rule via REST API), and 5 (patch branch protection rule via graphQL API), in favor of a single step where we create the entire branch protection rule with the graphQL API. 
All subsequent steps have been renumbered.

File: .github/workflows/updates.yaml (hunk 1 above)
Comment (gilzow) [reply .github/workflows/updates.yaml]: oh yeah, i need to go remove that comment. Will include it an upcoming push

**Comment by gilzow** [comment]
[fefd011](https://github.com/platformsh/template-builder/pull/462/commits/fefd011bbdbd860aaf62b6b798fa035aac5cef79) extends auto-update branch checking to see if the existing auto-update branch has been merged. Will only halt the process if it exists and either hasn't been merged, or has been merged and then advanced without an additional merge.

**Pull request merged**